draft-ietf-rats-eat.md

```diff
@@ -1099,12 +1103,53 @@ string naming the submodule. No submodules may have the same name.
 {::include cddl/submods.cddl}
 ~~~~
 
+
 # Endorsements and Verification Keys {#keyid}
 
 TODO: fill this section in. It will discuss key IDs, endorsement ID and such that
 are needed as input needed to by the Verifier to verify the signature. This will
 NOT discuss the contents of an Endorsement, just and ID/locator.
 
+
+# Layered Attestation {#layered}
+
+It is possible for a device to have multiple Attesting Environments as described in {{RATS-Architecture}}.
+When these are layered, one is a root the endorsement and/or verification key originates with the device manufacturer or endorser.
+The trust for those that are layered above the root comes from Attestation Evidence about them that is provided by the root or a layer below.
+
+A layered Attesting Environment is always described as a submodule.
+To know that a particular submodule is providing evidence for an Attesting Environment, it must have an intended use claim with the value of "attest=env".
+See {{intended-use}}.
+
+The submodule should also contain a claim that hold the verification key, the key used to verify signed attestation evidence from the layered attester.
+A specific claim is defined to hold verification keys.
+There is no requirement on the algorithm or key type used by the layered attestation.
+There is a requirement that the included key is in the format of a COSE_key.
+
+There are two ways that the Verifier comes to trust a layered Attesting Environment.
+
+First, the root or lower layer Attesting Environment provides claims, perhaps many claims, characterizing the Attesting Environment.
+The Verifier recognized the claims are about an Attesting Environment by the intended use claim and uses the other claims to decide whether it trusts this layered Attesting Environment.
+The decision is made by the Verifier.
+
+Second, the root or lower layer Attesting environment provides the is-trusted claim with a value of true.
+This tells the Verifier that the Attesting Environment has been sufficiently evaluated on the device by the root or lower layer Attesting Environment.
+The decision is made by on the device.
+
+Once the Verifier knows the Attesting Environment is trusted, then it makes use of the verification key to verify Attestation Evidence from the layered Attesting Environment.
+
+Multiple layers of Attesting Environments are allowed and must be represented as nested submodules.
+The layered Attesting Environment must be a submodule of the root of the token or a lower layer Attesting Environment.
+It doesn't have to be an immediate submodule, though. 
```

Comment:
I don't understand what that means. It probably means "The layered Attesting Environment" from the first sentence, but still I cannot really make sense of it.

Reply:
There are two ways to structure layers. (a) as nested entities or (b) as linked entities. Nesting has the undesirable side-effect that the signer (aka top layer) has to be in possession of all of the lower layer evidence in order to sign anything. This makes for ever increasing evidence blobs. Linking allows the upper layer to only sign the evidence it directly controls. The lower layer linked its evidence about the upper layer so there is no reason to counter-sign something that has already been signed by a lower layer.  Lets not think of submods as a hammer that forces nesting in ways that don't make sense.